### ISSUE TITLE
Ctrl+C exits `poetry shell` in Fish shell

### ISSUE THREAD

**Issue by jhrmnn** [sdispater/poetry#823]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.

- **OS version and name**: macOS 10.14.2
- **Poetry version**: 0.12.11

## Issue

When the Poetry shell is launched with `poetry shell` under the Fish shell, Ctrl+C exits the Poetry shell. This doesn't happen when launched from Bash. I would expect that Ctrl+C does nothing. I ended up not using `poetry shell`, but rather sourcing the `activate.fish` from the virtual environment, but that is a bit more cumbersome, because it's not directly accessible as a Poetry command.

**Comment by johnfraney** [comment]
I can reproduce this on Arch. With these versions:

poetry: 0.12.12
fish: 3.0.2

**Comment by actionless** [comment]
temporary workaround which i found:

```fish
function poetry_shell
        source $HOME/.cache/pypoetry/virtualenvs/(basename $PWD)-py3.7/bin/activate.fish
end
```

**Comment by johnfraney** [comment]
@jhrmnn This was fixed in [1.0.0b2](https://github.com/sdispater/poetry/releases/tag/1.0.0b2) by making the change I [mentioned above](https://github.com/sdispater/poetry/issues/823#issuecomment-486745360)
